packages/block-library/src/navigation/edit/utils.js

```diff
@@ -110,3 +120,10 @@ export function getNavigationChildBlockProps( innerBlocksColors ) {
 		},
 	};
 }
+
+export function getAllowedInnerBlocks() {
+	return applyFilters(
+		'blocks.navigation.__experimentalAllowedInnerBlocks',
```

Comment:
Is this is a filter everyone can use? It's a very specific filter. How do we handle allowed blocks for other blocks?

Reply:
FWIW, the motivation for this is a scenario where a user (e.g. accidentally) deletes a hooked block from a Navigation menu, and then wants to re-insert it -- in which case we'd want the hooked block to show up in the inserter, to which end it needs to be in the list of allowed blocks.

However, I agree that we should maybe remove this change from the PR. Hooked block insertion works fine without it (including editing), and it's an enhancement we can consider separately.

A client-side-only filter feels like an incomplete solution that cannot be extended later to cover the other need that Block Hooks have for a list of allowed inner blocks: Determining _on the server side_ whether a hooked block is actually allowed for a given anchor block, or if it should be skipped.

Reply:
As per my comment, removed in https://github.com/WordPress/gutenberg/pull/57754/commits/10f873b9d9696ee48468de7a03413a2a4311dbb9 😃

Reply:
Jarda kindly filed a PR to add an `allowedBlocks` attribute: https://github.com/WordPress/gutenberg/pull/58262

Reply:
Here's a dedicated issue on making Block Hooks respect the `allowedBlocks` field: https://github.com/WordPress/gutenberg/issues/53989